usdigitalresponse/usdr-gost
# Update emails to point to new Grant Details page

### Ticket #2644 
## Description
Updates the digest and assigned emails to link to our new Grant Details page when the flag is on, instead of directly to grants.gov. 

## Screenshots / Demo Video
<img width="1357" alt="Screenshot 2024-03-01 at 12 01 12 PM" src="https://github.com/usdigitalresponse/usdr-gost/assets/126250/e38f5136-cc4c-4076-aae6-d223b10d5fb1">
<img width="1357" alt="Screenshot 2024-03-01 at 12 04 17 PM" src="https://github.com/usdigitalresponse/usdr-gost/assets/126250/bb98ae46-7a82-4be7-8037-f6cb90f6a5e2">

## Testing
Testing this in dev is a lot of pretty fiddly effort, but here are the steps: 

1. Set up so emails actually get sent/captured in dev
   1. Create an ethereal email to work with: https://ethereal.email/ (nice, one-click creation of an ephemeral email account)
   2. Update the `NODEMAILER_*` environment variables in your `packages/server/.env` file from the values listed on your ethereal account
   3. Comment out `secure: true` in our nodemailer setup (Gmail uses a secure connection, but ethereal doesn't): https://github.com/usdigitalresponse/usdr-gost/blob/05f688e75df1e3fbf6698f88b49281dd117eb280/packages/server/src/lib/email/email-nodemailer.js#L36
   4. Rebuild your docker container so it picks up the new env variables (`docker compose down app` and `docker compose up -d`)
2. Trigger the emails manually via a node REPL
   - Digest email
     1. `docker compose exec app yarn node`
     2. ```node
           const emailService = require('./packages/server/src/lib/email');
           const knex = require('./packages/server/src/db/connection')
           const grants = await knex('grants').limit(3)
           emailService.sendGrantDigest({name: 'Test agency', matchedGrants: grants, recipients: ['test@example.com'], openDate: '2024-01-01'})
           ```
   - Assignment email
     1. Comment out the `assignee` variable in `sendGrantAssignedNotficationForAgency` and replace it with `const assignees = [{ email: 'test@example.com' }];`: https://github.com/usdigitalresponse/usdr-gost/blob/05f688e75df1e3fbf6698f88b49281dd117eb280/packages/server/src/lib/email.js#L221
     2. `docker compose exec app yarn node`
     3. ```node
           const emailService = require('./packages/server/src/lib/email');
           const knex = require('./packages/server/src/db/connection')
           const grant = await knex('grants').first();
           emailService.sendGrantAssignedEmail({ grantId: grant.grant_id, agencyIds: [1], userId: 1 });
           ```
3. Once the test email is sent, go to your "Messages" tab in your Ethereal mail account, and you should see the sent email listed there for viewing

### Automated and Unit Tests

### Manual tests for Reviewer

## Checklist
